### PR TITLE
MVM: Define MVM-specific processing parameters for drizzling

### PR DESCRIPTION
**Pull request by dulude**
**Related JIRA tickets and git issues**
- [HLA-710](https://jira.stsci.edu/browse/HLA-710)
- [HLA-713](https://jira.stsci.edu/browse/HLA-713)
- #1257 

**Other associated JIRA tickets and git issues**
- [HLA-714](https://jira.stsci.edu/browse/HLA-714)
- #1261 

**Summary of changes**
- The ultimate goal of these tickets were to add MVM-specific astrodrizzle parameter files  any_n1.json and any_n2.json for ACS/HRC, ACS/WFC, WFC3/IR and WFC3/UVIS.
- To this end, a number of changes had to be made:
  - Separate "mvm_parameters" and "svm_parameters" parameter directories have replaced the "default_parameters" directory
  - MVM and SVM-specific versions of the pars/hap_pars/*index.json files have been created and are now found in "svm_parameters" and "mvm_parameters".
  - Added new MVM-specific instrument/detector-specific any_n1.json astrodrizzle parameter files for ACS/HRC, ACS/WFC, WFC3/IR and WFC3/UVIS created by @mackjenn and @rjavila for ticket [HLA-713](https://jira.stsci.edu/browse/HLA-713) 
  - Updated MVM-specific astrodrizzle any_n2.json astrodrizzle parameter files for ACS/HRC, ACS/WFC, WFC3/IR and WFC3/UVIS with values stored in any_n2.json files created by @mackjenn and @rjavila for ticket [HLA-713](https://jira.stsci.edu/browse/HLA-713) 
  - updated config_utils.py to function properly with the newly updated parameter directory structure. 
    - Added a new optional input "hap_pipeline_name" facilitate the new functionality.
    - updated upstream calls to config_utils.py in hapsequencer.py, hapmultisequencer.py and generate_custom_svm_param_file.py
    - updated HapConfig._determine_conditions() to use the "any_n1" condition for MVM runs processing a single lonely exposure, and the "any_n2" conditions for MVM runs processing any number of exposures greater than 1.
  - centralized automatic update of "quality control" params "ci_lower_limit" and "ci_upper_limit" from lookup table file with the addition of method config_utils.HapConfig._update_ci_values_from_file() so that these parameters are all updated at the time of instantiation.
- Corrected issue found both MVM an SVM versions of the acs_hrc_index.json file. None of the conditions set in config_utils.HapConfig._determine_conditions for n_exposures > 1 match the condition/parameter file mappings in the  acs_hrc_index.json files. The possible conditions generated by _determine_conditions() for ACS/HRC are "acs_hrc_any_n2", "acs_hrc_any_n4" and "acs_hrc_any_n6". However, conditions defined in acs_hrc_index.json files are "acs_any_post_n2", "acs_any_post_n4" and "acs_any_post_n6". To fix this, the "post" was removed from conditions in the hrc index files.

### CODE REVIEW

**Review by stsci-hack — CHANGES_REQUESTED** [review]
This looks pretty good, but there were a couple of small things that need to be addressed before this can be merged.

**Comment by stsci-hack** [comment]
Also, the failures are due to missing a new parameter in a call to the config parameters functions that are being updated in this PR, as shown in this traceback.  This also needs to be fixed before it can be merged.

```
2022-02-24T16:15:13.5955557Z /home/vsts/work/1/s/drizzlepac/runastrodriz.py:566: in process
2022-02-24T16:15:13.5956225Z     align_aposteriori, aposteriori_table = verify_alignment(_inlist,
2022-02-24T16:15:13.5956909Z /home/vsts/work/1/s/drizzlepac/runastrodriz.py:942: in verify_alignment
2022-02-24T16:15:13.5957541Z     instdet_pars = align.get_default_pars(inst, det)
2022-02-24T16:15:13.5958141Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-02-24T16:15:13.5958485Z 
2022-02-24T16:15:13.5959134Z instrument = 'wfc3', detector = 'uvis', step = 'alignment'
2022-02-24T16:15:13.6272766Z condition = ['filter_basic']
2022-02-24T16:15:13.6274586Z 
2022-02-24T16:15:13.6275869Z     def get_default_pars(instrument, detector, step='alignment',
2022-02-24T16:15:13.6276928Z                          condition=['filter_basic']):
2022-02-24T16:15:13.6277564Z     
2022-02-24T16:15:13.6278130Z         step_list = config_utils.step_title_list
2022-02-24T16:15:13.6278758Z         if step not in step_list:
2022-02-24T16:15:13.6280042Z             log.error("{} not valid!  Needs to be one of: {}".format(step,
2022-02-24T16:15:13.6280716Z                       step_list))
2022-02-24T16:15:13.6281289Z             raise ValueError
2022-02-24T16:15:13.6281826Z     
2022-02-24T16:15:13.6282488Z >       full_cfg_index, pars_dir = config_utils.read_index(instrument, detector)
2022-02-24T16:15:13.6283631Z E       TypeError: read_index() missing 1 required positional argument: 'detector'
2022-02-24T16:15:13.6284171Z 
```

File: drizzlepac/hapmultisequencer.py [hunk 1]
```diff
@@ -377,19 +377,22 @@ def run_mvm_processing(input_filename, skip_gaia_alignment=True, diagnostic_mode
...
                 edp.configobj_pars = config_utils.HapConfig(edp,
+                                                            hap_pipeline_name='mvm',
                                                             log_level=log_level,
                                                             use_defaults=use_defaults_configs,
                                                             input_custom_pars_file=input_custom_pars_file,
                                                             output_custom_pars_file=output_custom_pars_file)
         log.info("The configuration parameters have been read and applied to the drizzle objects.")
-
+        import pdb
+        pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (dulude) [reply drizzlepac/hapmultisequencer.py]: Resolved by commit [e70746f](https://github.com/spacetelescope/drizzlepac/pull/1277/commits/e70746f840fe1f4d1f0b546e366871620fd69b5d)

File: drizzlepac/haputils/config_utils.py [hunk 2]
```diff
@@ -178,30 +208,37 @@ def _determine_conditions(self, prod_obj):
...
+                        if self.hap_pipeline_name == 'mvm':
+                            self.conditions.append("wfc3_uvis_any_post_n2")
+                        if self.hap_pipeline_name == 'svm':
+                            thresh_time = Time("2012-11-08T02:59:15", format='isot', scale='utc').mjd
+                            # Get the MJDUTC of the first exposure in the filter exposure product list. While
+                            # each exposure will have its own MJDUTC (the EXPSTART keyword), this is probably
+                            # granular enough.
+                            mjdutc = prod_obj.edp_list[0].mjdutc
+                            if mjdutc >= thresh_time:
+                                if n_exp in [2, 3]:
```
[the remark this answers is not in the document]
Comment (dulude) [reply drizzlepac/haputils/config_utils.py:235]: Resolved by commit [56283c1](https://github.com/spacetelescope/drizzlepac/pull/1277/commits/56283c1b17f546648a889d8cedfd309ff94805ac)

**Review by stsci-hack — APPROVED** [review]
Tests pass.  Looks like this works as expected now.

**Pull request merged**